### PR TITLE
Add TCH001 and TCH002 rules to pre-commit to detect and move type checking modules

### PR DESCRIPTION
**Pull request by hussein-awala**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by hussein-awala** [comment]
@potiuk do we cherry-pick dev and tests? is it ok to keep all these changes in the same PR or should I split them in smaller ones?

File: pyproject.toml [hunk 1]
```diff
@@ -62,7 +62,10 @@ extend-select = [
     "D402",
     "D403",
     "D412",
-    "D419"
+    "D419",
+    # type-checking
```
Comment (uranusjr) [inline pyproject.toml]: This comment is pretty useless tbh

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py [hunk 2]
```diff
@@ -39,6 +36,8 @@
     # We handle the ImportError so that autocomplete works with just click installed
     version = None  # type: ignore[assignment]
 
+from typing import TYPE_CHECKING
```
Comment (uranusjr) [inline dev/breeze/src/airflow_breeze/utils/docker_command_utils.py]: This import seems misplaced

File: pyproject.toml [hunk 3]
```diff
@@ -62,7 +62,9 @@ extend-select = [
     "D402",
     "D403",
     "D412",
-    "D419"
+    "D419",
+    "TCH001", # typing-only-first-party-import
+    "TCH002", # typing-only-third-party-import
```
Comment (uranusjr) [inline pyproject.toml]:
```suggestion
    "TCH001",  # typing-only-first-party-import
    "TCH002",  # typing-only-third-party-import
```

Not actually revelant but this matches how Black formats Python code

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
> @potiuk do we cherry-pick dev and tests? is it ok to keep all these changes in the same PR or should I split them in smaller ones?

We do cherry-pick dev and tests except providers (we might cherry-pick some of them accidentally) -  but I think with the current state of 2.7 this does not matter any more :D

**Pull request merged**